File: src/ios/trees/Site.swift
import Foundation

class Site: QtElement
{
    init(json: JsonObject, species: Species)
    {
        self.location = QtLocation(json: json)
        self.species = species
    }

    init(location: QtLocation, species: Species)
    {
        self.location = location
        self.species = species
    }

    public let location: QtLocation
    public let species: Species
}

File: src/android/app/src/main/java/com/crickettechnology/trees/Site.java
package com.crickettechnology.trees;

import org.json.*;

class Site implements QtElement
{
    public Site(JSONObject json, Species species)
    {
        double latitude = json.optDouble("latitude");
        double longitude = json.optDouble("longitude");
        this.location = new QtLocation(latitude, longitude);
        this.species = species;
    }

    public Site(QtLocation location, Species species)
    {
        this.location = location;
        this.species = species;
    }

    public final QtLocation location;
    public final Species species;

    ////////////////////////////////////////
    // QtElement

    public QtLocation getLocation() { return this.location; }

}
